sdk/js/src/token_bridge/attest.ts

```diff
@@ -197,3 +203,81 @@ export async function attestFromAlgorand(
 
   return txs;
 }
+
+/**
+ * Attest an already created asset
+ * If you create a new asset on near and want to transfer it elsewhere,
+ * you create an attestation for it on near... pass that vaa to the target chain..
+ * submit it.. then you can transfer from near to that target chain
+ * @param client An Near account client
+ * @param coreBridge The account for the core bridge
+ * @param tokenBridge The account for the token bridge
+ * @param asset The account for the asset
+ * @returns [sequenceNumber, emitter]
+ */
+export async function attestTokenFromNear(
+  client: nearAccount,
+  coreBridge: string,
+  tokenBridge: string,
+  asset: string
+): Promise<[number, string]> {
+  let message_fee = await client.viewFunction(coreBridge, "message_fee", {});
+  // Non-signing event
+  if (!getIsWrappedAssetNear(tokenBridge, asset)) {
+    // Non-signing event that hits the RPC
+    let res = await client.viewFunction(tokenBridge, "hash_account", {
+      account: asset,
+    });
+
+    // if res[0] == false, the account has not been
+    // registered... The first user to attest a non-wormhole token
+    // is gonna have to pay for the space
+    if (!res[0]) {
+      // Signing event
+      await client.functionCall({
+        contractId: tokenBridge,
+        methodName: "register_account",
+        args: { account: asset },
+        gas: new BN("100000000000000"),
+        attachedDeposit: new BN("2000000000000000000000"), // 0.002 NEAR
+      });
+    }
+  }
+
+  // Signing event
+  let result = await client.functionCall({
+    contractId: tokenBridge,
+    methodName: "attest_token",
+    args: { token: asset, message_fee: message_fee },
+    attachedDeposit: new BN("3000000000000000000000") + new BN(message_fee), // 0.003 NEAR
+    gas: new BN("100000000000000"),
+  });
+
+  return parseSequenceFromLogNear(result);
+}
+
+/**
+ * Attest NEAR
+ * @param client An Near account client
+ * @param coreBridge The account for the core bridge
+ * @param tokenBridge The account for the token bridge
+ * @returns [sequenceNumber, emitter]
+ */
+export async function attestNearFromNear(
+  client: nearAccount,
+  coreBridge: string,
+  tokenBridge: string
+): Promise<[number, string]> {
+  let message_fee =
+    (await client.viewFunction(coreBridge, "message_fee", {})) + 1;
```

Comment:
Just out of curiosity, why +1?

Reply:
We want at least 1 passed in so that this is a "signed event" in the case there is a 0 message_fee.  It is a strange security thing in Near.  If you don't attach a deposit, you could actually make the call without authentication...  So, people will require at least 1 yocto of Near be attached so that we have a env::signer ...